docs/upgrade.md

```diff
@@ -3,5 +3,261 @@ title: Upgrade guide
 description: Guide to update between major Powertools for AWS Lambda (TypeScript) versions
 ---
 
-!!! warning
-    This guide is a work in progress. We'll update it as we get closer to the 2.0 release. If you have any questions, or want to follow the progress, please join the discussion on the [GitHub issue](https://github.com/aws-powertools/powertools-lambda-typescript/issues/1714).
\ No newline at end of file
+## Migrate from v1 to v2
+
+We've made targeted breaking changes to make your transition to v2 as smooth as possible while still providing you with the most up-to-date features and patterns. This guide will help you migrate your existing codebase to v2.
+
+### Quick summary
```

Comment:
Should we also add a note about layer changes, so customers need to use new layer name? There might be expectation that we only increment the version number.

Reply:
I think that's covered by the section below (First steps) where we say "ensure that you have the latest version via Lambda Layer" with the section in the docs linked. 

That section in the docs will have the new ARN.